kafka/metrics.go

```diff
@@ -0,0 +1,103 @@
+package kafka
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+const instrumentName = "github.com/elastic/apm-queue/kafka"
+const unitCount = "1"
+
+type instruments struct {
+	MessageProduced metric.Int64Counter
+	WriteErrors     metric.Int64Counter
+	WriteTimeout    metric.Int64Counter
```

Comment:
How about making this a `counters map[string]metric.Int64Counter`?
Then, you can store each generated metric within the map, and retrieve them by calling the proper key. And when we add more metrics, we only need to instantiate them, not to add new parameters to the struct.

Example: https://github.com/elastic/apm-ingest-service/pull/188/files#diff-8dcf09af3338b5e3276cbcefae7f6b12b8a4bde0662c242317cc543af8f95d15R63
And in otel's net/http: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/net/http/otelhttp/handler.go#L46

Reply:
I'm not sure I agree. Seems unnecessary to add an unstructured map here since the metrics are well defined.

Reply:
I agree with @marclop, I don't see a good reason to use a map when we could use specific fields. The latter approach will naturally have lower overhead.